File: src/main/java/net/heartshapedbox/math/Quad.java
package net.heartshapedbox.math;

import net.minecraft.util.math.Vec3d;

public class Quad {
    public Vec3d[] points = new Vec3d[4];
    
    public Quad(Vec3d point1, Vec3d point2, Vec3d point3, Vec3d point4) {
        points[0] = point1;
        points[1] = point2;
        points[2] = point3;
        points[3] = point4;
    }
}

File: src/main/kotlin/io/github/faecraft/heartshapedbox/math/Quad.kt
package io.github.faecraft.heartshapedbox.math

import io.github.faecraft.heartshapedbox.util.QuadSame
import net.minecraft.util.math.Vec3d

public data class Quad(val point1: Vec3d, val point2: Vec3d, val point3: Vec3d, val point4: Vec3d) {
    val points: QuadSame<Vec3d> = QuadSame(point1, point2, point3, point4)
}
